File: EiMT_lab/backend/src/main/java/finki/ukim/emt/booking/model/domain/Accommodation.java
package finki.ukim.emt.booking.model.domain;

import finki.ukim.emt.booking.model.enums.Category;
import finki.ukim.emt.booking.model.enums.Condition;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

@Entity
@Getter
@Setter
@NoArgsConstructor
@Table(name = "accommodations")
public class Accommodation extends BaseAuditableEntity {
    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private Category category;

    @ManyToOne
    @OnDelete(action = OnDeleteAction.CASCADE)
    @JoinColumn(name = "host_id", nullable = false)
    private Host host;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private Condition condition = Condition.GOOD;

    @Column(nullable = false)
    private Integer numRooms;

    @Column(nullable = false)
    private Boolean rented = Boolean.FALSE;

    public Accommodation(String name, Category category, Host host, Condition condition, Integer numRooms) {
        this.name = name;
        this.category = category;
        this.host = host;
        this.condition = condition;
        this.numRooms = numRooms;
        this.rented = false;
    }

    public Accommodation(String name, Category category, Host host, Condition condition, Integer numRooms, Boolean rented) {
        this.name = name;
        this.category = category;
        this.host = host;
        this.condition = condition;
        this.numRooms = numRooms;
        this.rented = rented;
    }
}

File: EiMT_lab/frontend/src/api/types/accommodation.ts
export type Category = 'ROOM' | 'HOUSE' | 'FLAT' | 'APARTMENT' | 'HOTEL' | 'MOTEL';
export type Condition = 'GOOD' | 'BAD';

export interface Accommodation {
  id: number;
  name: string;
  category: Category;
  condition: Condition;
  hostId: number;
  hostFullName: string;
  numRooms: number;
  rented: boolean;
}

export interface AccommodationDetails {
  id: number;
  name: string;
  category: Category;
  condition: Condition;
  hostId: number;
  hostFullName: string;
  numRooms: number;
  rented: boolean;
}

export interface CreateAccommodationRequest {
  name: string;
  category: Category;
  hostId: number;
  condition: Condition;
  numRooms: number;
}
